Source/cmake/OptionsGTK.cmake

```diff
@@ -19,7 +19,7 @@ find_package(Freetype 2.4.2 REQUIRED)
 find_package(LibGcrypt 1.6.0 REQUIRED)
 find_package(GLIB 2.56.4 REQUIRED COMPONENTS gio gio-unix gobject gthread gmodule)
 find_package(HarfBuzz 0.9.18 REQUIRED COMPONENTS ICU)
-find_package(ICU 61.2 REQUIRED COMPONENTS data i18n uc)
+find_package(ICU 60.1 REQUIRED COMPONENTS data i18n uc)
```

Comment:
I don't agree to this change until we get a consensus that,

1. Intl implementators and upstream do not care about old ICU
2. The maintainer of downstream port needs to care (they need to monitor and fix).

I'm OK to having a patch supporting ICU 60. But I don't want to be involved into the support of this old ICU. ICU 60 support should be done by maintainers who care about ICU 60, not by Intl developers.

Reply:
> where we don't have any new bots and only downstream maintainers (i.e. me) are responsible for keeping the build working. However, it would still be a bug if it doesn't build, so you wouldn't _intentionally_ commit anything that breaks the oldest supported version. Does that sound OK?

I don’t think this is the right place to debate this, but …

As a contributor the above sounds like torture to me. I really don’t like things that I am supposed to do, but with no way to check. Knowing that there are people getting annoyed at me for making mistakes I can’t easily avoid. It’s kind of the opposite of the good feeling of contributing to a project for me.

Reply:
There's not really much we can do about that, though. We don't have EWS bots with the minimum theoretical version of everything installed. That would be too much effort to maintain. Instead, we have post-commit LTS bots that may not be ideal, but work well enough in practice. We don't blame you if they break because they're not EWS so you can't really know. :)

Reply:
I suspect what @Constellation is most worried about is when a contributor wants to use something we hadn’t previously used in ICU, and in such cases we want to remove old code paths that now aren’t needed. Over the last few years, we have gained a lot by moving forward on ICU use, and the specter of not being able know if we can do that might not be great for contributors working in these areas.

Reply:
Keeping old ICU support is *not* a trivial task. ICU is constantly extending project. ICU is adding new API and new interface / class to replace the old one. And even the old one gets deprecated later. Supporting old ICU means that it prevents our code base from refactoring because we need to maintain two classes. For example,

1. IntlNumberFormat literally has two versions of implementation because `UNumberFormatter` is available only for the newer version of ICU, and it is replacing the old `UNumberFormat`.
2. We are maintaining `UFieldPositionIterator` old use of field iteration because new version of ICU API (UFormattedValue) is not available in old ICU. We need to use `UFormattedValue` based enumeration for new ICU APIs since it is used in them. And for old ICU APIs, while `UFormattedValue` is supported, we need to use `UFieldPositionIterator` since `UFormattedValue` is not available for old ICU.

Furthermore, we need draft (non stable!) ICU API for new Intl features. And they can even change their behavior based on ICU version since it is draft API, and actually, this is happening. Intl developers are actively watching on ICU development, tracking changes in some non-compile-failure changes (number format skeleton syntax for example). You can actually see that we are changing skeleton construction based on runtime ICU library version (since header version is not reliable).

For example,
```cpp
  1480     // After ICU 67, udtitvfmt_formatToResult's signature is changed.
  1481 #if U_ICU_VERSION_MAJOR_NUM >= 67
  1482     // If a date is after Oct 15, 1582, the configuration of gregorian calendar change date in UCalendar does not affect
  1483     // on the formatted string. To ensure that it is after Oct 15 in all timezones, we add one day to gregorian calendar
  1484     // change date in UTC, so that this check can conservatively answer whether the date is definitely after gregorian
  1485     // calendar change date.
  1486     auto definitelyAfterGregorianCalendarChangeDate = [](double millisecondsFromEpoch) {
  1487         constexpr double gregorianCalendarReformDateInUTC = -12219292800000.0;
  1488         return millisecondsFromEpoch >= (gregorianCalendarReformDateInUTC + msPerDay);
  1489     };
  ```

```cpp
  376         // 3.k. If style is "2-digit", then
  377         //     i. Perform ! CreateDataPropertyOrThrow(nfOpts, "minimumIntegerDigits", 2F).
  378         skeletonBuilder.append(" integer-width/", WTF::ICU::majorVersion() >= 67 ? '*' : '+'); // Prior to ICU 67, use the symbol + instead of *.
  ```
  
  Moreover, ICU is constantly changing CLDR data for every version. This affects on the output of ICU formatted data, and thus, it is very hard to keep tests work with various ICU versions. Even Apple and Google ICU CLDR are modified (AppleICU has overrides for CLDR. Google Chrome's ICU also overrides CLDR (arab numbering system for example). So, keeping tests run and adding new tests incur cost to ensure that whether we can list up patterns work in various ICU versions.
  
  There are more things like this, and this kind of issue will increase more.

We are paying this huge cost because these versions are cared by Intl developers.
But this ICU 60 is too old, and no Intl developers are maintaining this ICU version right now.
Adding ICU 60 support means,

1. More and more ifdefs remains and we need to keep many duplicate things with different APIs for old ICU, code and architecture health suffered significantly (thus, productivity is regressed too). https://trac.webkit.org/wiki/WebKitGTK/DependenciesPolicy allows us to drop and keep Intl implementation sane. But adding ICU 60 support keeps them.
2. For new API usage with new interface, we need to write the duplicate code which should be replaced in new ICU. And need to keep new and old versions and need to make both of them tested.
3. Need to write tests which can work in older ICUs, which makes writing tests more complicated.

So I'm saying here that please do not increase maintenance burden significantly more and do not hurt Intl development's productivity. Resource of Intl developers are limited, and ICU 60 support hurts productivity and code / architecture health.

Reply:
I'll pass this along to Red Hat, but there's a 0% chance we will ever update ICU due to the ABI breaks. So when it becomes too hard to maintain the ICU support, we'll just stop updating WebKitGTK.

Reply:
If the system cannot upgrade that very old ICU, then bundling ICU is the right path (as the other browsers do).

Reply:
We did that for RHEL 7 but I don't want to do it again. It's a mess. I prefer to patch WebKit for as long as possible instead.

Reply:
BTW if your concerns are primarily about the Intl API, could we just add a feature flag to disable the whole Intl API? Then we don't have to worry about this at all, and for good measure I could stop reporting bugs about Intl test failures....

That won't help with `UFieldPositionIterator` though.

Reply:
This makes things complicated. We have many things which rely on existence of sane version of ICU. And more and more Intl things are coupled with the rest of the ECMAScript standards recently. For example, Temporal proposal is tied with Intl.

Reply:
I feel bundling ICU is the right path if you cannot update the system ICU.
Chromium and Firefox do that. And since WTF is linked statically to JSC, then you can construct the Intl utilities in WTF and then you can statically link ICU to JSC.

Reply:
Again, I'll continue patching WebKit for as long as possible rather than static link to ICU.